src/settings/toml/migrations.rs

```diff
@@ -0,0 +1,499 @@
+            new_classes.extend(mc.migration.durable_objects.new_classes.iter().cloned());
+            transferred_classes.extend(
+                mc.migration
+                    .durable_objects
+                    .transferred_classes
+                    .iter()
+                    .cloned(),
+            );
+        }
+        Ok(ApiMigration {
+            old_tag: old_tag.map(|t| t.to_owned()),
+            new_tag: migrations.last().map(|c| c.tag.clone()).flatten(),
+            migration: Migration {
+                durable_objects: DurableObjectsMigration {
+                    new_classes,
+                    deleted_classes,
+                    renamed_classes,
+                    transferred_classes,
+                },
+            },
+        })
+    }
+}
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize)]
+struct ListScriptsV4ApiResponse {
+    pub result: Vec<ApiScript>,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize)]
+struct ApiScript {
+    #[serde(rename = "id")]
+    pub name: String,
+    pub migration_tag: Option<String>,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct MigrationConfig {
+    pub tag: Option<String>,
+    #[serde(flatten)]
+    pub migration: Migration,
+}
+
+impl MigrationConfig {
+    pub fn add_to_app<'a, 'b>(app: App<'a, 'b>) -> App<'a, 'b> {
+        app
+        .arg(
+            Arg::with_name("new_class")
+            .help("allow durable objects to be created from a class in your script")
+            .long("new-class")
+            .takes_value(true)
+            .number_of_values(1)
+            .value_name("class name")
+            .multiple(true)
+        )
+        .arg(
+            Arg::with_name("delete_class")
+            .help("delete all durable objects associated with a class in your script")
+            .long("delete-class")
+            .takes_value(true)
+            .number_of_values(1)
+            .value_name("class name")
+            .multiple(true)
+        )
+        .arg(
+            Arg::with_name("rename_class")
+            .help("rename a durable object class")
+            .long("rename-class")
+            .takes_value(true)
+            .number_of_values(2)
+            .value_names(&["from class", "to class"])
+            .multiple(true)
+        )
+        .arg(
+            Arg::with_name("transfer_class")
+            .help("transfer all durable objects associated with a class in another script to a class in this script")
+            .long("transfer-class")
+            .takes_value(true)
+            .number_of_values(3)
+            .value_names(&["from script", "from class", "to class"])
+            .multiple(true)
+        )
+        .group(
+            ArgGroup::with_name("migrations")
+            .args(&["new_class", "delete_class", "rename_class", "transfer_class"])
+            .multiple(true)
+        )
```

Comment:
dude this pattern is so good...we should make this a trait and impl for subcommands in a future pr